presto-hive-common/src/main/java/com/facebook/presto/hive/zorder/ZOrder.java

```diff
@@ -191,6 +194,42 @@ public long zOrderByteAddressToLong(byte[] byteAddress)
         return byteAddress;
     }
 
+    /**
+     * Searches for and outputs ranges of long addresses within certain ranges in each dimension.
+     */
+    public List<ZAddressRange> zOrderCondensedSearchCurveLongs(List<ZValueRange> ranges)
+    {
+        curveSearcher.setReturnType(ZOrderCurveSearcher.ReturnType.LONG);
+        return curveSearcher.zOrderCondensedSearchCurve(ranges);
+    }
+
+    /**
+     * Searches for and outputs the long addresses within certain ranges in each dimension.
+     */
+    public List<Long> zOrderSearchCurveLongs(List<ZValueRange> ranges)
+    {
+        curveSearcher.setReturnType(ZOrderCurveSearcher.ReturnType.LONG);
+        return curveSearcher.zOrderSearchCurveLongs(ranges);
+    }
+
+    /**
+     * Searches for and outputs ranges of integer addresses within certain ranges in each dimension.
+     */
+    public List<ZAddressRange> zOrderCondensedSearchCurveIntegers(List<ZValueRange> ranges)
+    {
+        curveSearcher.setReturnType(ZOrderCurveSearcher.ReturnType.INTEGER);
```

Comment:
Let's do static import and also directly pass it as an argument for `zOrderCondensedSearchCurve`? I don't think we need this setter method.